go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -623,12 +623,27 @@ func generateEIPConfig(link netlink.Link, eIPNet *net.IPNet, isEIPV6 bool) (*eIP
 }
 
 func generateRoutesForLink(link netlink.Link, isV6 bool) ([]netlink.Route, error) {
-	linkRoutes, err := netlink.RouteList(link, util.GetIPFamily(isV6))
+	routeTable := 254 // main table number
+	// check if device is a slave to a VRF device and if so, use VRF devices associated routing table to lookup routes instead of main table
+	if isVRFSlaveDevice(link) {
+		vrfLink, err := util.GetNetLinkOps().LinkByIndex(link.Attrs().MasterIndex)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get VRF link from interface index %d: %v", link.Attrs().MasterIndex, err)
+		}
+		vrf, ok := vrfLink.(*netlink.Vrf)
+		if !ok {
+			return nil, fmt.Errorf("expected link %s to be VRF", vrfLink.Attrs().Name)
+		}
+		routeTable = int(vrf.Table)
```

Comment:
In the kernel the `table_id` is a `u32`, meaning that this conversion to `int` _technically_ (extra, EXTRA emphasis on _technically_) could overflow and wrap around on a 32-bit system if the VRF table_id is set really high for some reason? I don't think ovn-kubernetes claims to support 32-bit systems though, so I expect this is a non-issue?